### PR TITLE
feat: introduce and implement `--rename-symbol`

### LINKED ISSUE #5419: Need command-line option to contextually rename symbols
For our LSP integration, we will need a command-line option to contextually rename symbols. It would best if we could rename them across a project rather than just the current file, but we must be able to rename them within at least the current file.

Support for this will require indexing a symbol's locations within the current file and possibly within its project. We may then use that same index for LSP's various symbol reference functionalities.

Requirements:
1. Need a command-line option, named `--rename-symbol` or the like.
2. It will accept the following parameters:
    1. `--line` := line on which the symbol to rename appears (accepts int param; must be within the respective range)
    2. `--column` := column on which the symbol to rename appears (accepts int param; must be within the respective range)
    3. `--new-name` (or something like that) := what to replace the symbol with (accepts string param; must be valid identifier)
    4. The final, positional parameter will be the path to the file whose symbol should be renamed (as is the current behavior).

### PR DESCRIPTION
**Pull request by Pranavchiku**
Fixes #5419

### CODE REVIEW

File: src/libasr/lsp.cpp [hunk 1]
```diff
@@ -9,7 +9,7 @@
 #include <libasr/location.h>
 #include <lfortran/fortran_evaluator.h>
 #include <libasr/lsp_interface.h>
-
+#include "libasr/asr_lookup_name.cpp"
```
Comment (certik) [inline src/libasr/lsp.cpp:12]: What is going on here? We should not be including `.cpp` files.

File: src/libasr/lsp.cpp [hunk 2]
```diff
@@ -391,5 +391,88 @@ int get_definitions(const std::string &infile, LCompilers::CompilerOptions &comp
...
+
+        range_object.SetObject();
+
+        start_detail.SetObject();
+        start_detail.AddMember("character", start_character);
+        start_detail.AddMember("line", start_line);
+        range_object.AddMember("start", start_detail);
+
+        end_detail.SetObject();
+        end_detail.AddMember("character", end_character);
```
Comment (certik) [inline src/libasr/lsp.cpp:457]: Let's call this `column`?

File: src/libasr/lsp.cpp (hunk 1 above)
Comment (Pranavchiku) [reply src/libasr/lsp.cpp:12]: We need this for `OccurenceCollector` visitor. It is implemented in `libasr/asr_lookup_name.cpp`

File: src/libasr/lsp.cpp (hunk 2 above)
Comment (Pranavchiku) [reply src/libasr/lsp.cpp:457]: We can, but then we need to change handling at `lfortran-lsp`

Comment (certik) [reply src/libasr/lsp.cpp:457]: Ok. We can do that later.

File: src/libasr/lsp.cpp (hunk 1 above)
Comment (certik) [reply src/libasr/lsp.cpp:12]: Ok.

**Comment by Pranavchiku** [comment]
Is this good to merge?

File: src/libasr/lsp.cpp (hunk 1 above)
Comment (certik) [reply src/libasr/lsp.cpp:12]: Can you implement it in a header file? I think it's bad form to be including `.cpp` files. Maybe you can just rename `asr_lookup_name.cpp` to `asr_lookup_name.h`.

**Review by certik — APPROVED** [review]
I think that's fine. But if you can fix the `cpp` issue based on my comment above, that would be great. You can do it in a subsequent PR.

File: src/libasr/asr_lookup_name.h [hunk 3]
```diff
@@ -0,0 +1,140 @@
+#include <iostream>
```
Comment (certik) [inline src/libasr/asr_lookup_name.h:1]: You have to put guards into the header file.

**Comment by certik** [comment]
The error is just a temporary error I think:
```
  Transaction starting
  critical libmamba Download error (28) Timeout was reached [https://conda.anaconda.org/conda-forge/osx-arm64/llvm-tools-11.1.0-hfa12f05_5.tar.bz2]
      Operation too slow. Less than 30 bytes/sec transferred the last 60 seconds
  Error: The process '/Users/runner/micromamba-bin/micromamba' failed with exit code 1
```

**Review by certik — APPROVED** [review]

**Review by certik — APPROVED** [review]
I think this looks great, thanks!

**Pull request merged**